File: backend/app/domain/entities/rol.py
"""Entidad de dominio Rol (RF-028, RF-029)."""
from dataclasses import dataclass, field

from app.domain.entities.permiso import Permiso


@dataclass
class Rol:
    nombre: str
    descripcion: str | None = None
    es_sistema: bool = False
    permisos: list[Permiso] = field(default_factory=list)
    id: int | None = None

File: frontend/src/entities/rol.ts
export interface Permiso {
  id: number
  modulo: string
  accion: string
  codigo: string
  descripcion: string | null
}

export interface Rol {
  id: number
  nombre: string
  descripcion: string | null
  es_sistema: boolean
  permisos: Permiso[]
}
